src/asset_manager/view/AssetImageView.js

```diff
@@ -1,9 +1,9 @@
 module.exports = require('./AssetView').extend({
 
   events: {
+    'click [data-toggle=asset-remove]': 'onRemove',
```

Comment:
I couldn't find it documented anywhere, but these events seem to fire in the order they are registered. The onRemove function needs to run before onClick so that the event propagation can be stopped.